plugins/detabler/plugin.js

```diff
@@ -1,9 +1,69 @@
 'use strict';
 
-(function() {
-    CKEDITOR.plugins.add( 'detabler', {    
-    init: function( editor ) {
-        // Plugin logic goes here...
-    }
-});
-})();
\ No newline at end of file
+(function () {
+    CKEDITOR.plugins.add('detabler', {
+		requires: 'table,contextmenu',
+		init: function (editor) {
+			editor.addCommand('detable', {
+				exec: function (editor) {
+					var path,
+						table,
+						cells,
+						currentCell,
+						currentItem;
+
+					path = editor.elementPath();
+					table = path.contains('table');
+
+					if (table) {
```

Comment:
To save some indentation, what do you think about doing `if (!table) { return; }` or something else equivalent?